### ISSUE TITLE
Find a way to check power usage

### ISSUE THREAD

**Issue by JakeRoggenbuck** [JakeRoggenbuck/auto-clock-speed#41]

**Comment by Camerooooon** [comment]
Is this checking something like the voltage being pulled? Or the CPU usage?

**Comment by JakeRoggenbuck** [comment]
I think it would be cool to know how much power your cpu is using, and then how much battery percentage is being used in general

**Comment by Camerooooon** [comment]
I'm on it

**Comment by JakeRoggenbuck** [comment]
Awesome!!

**Comment by Camerooooon** [comment]
I cannot find a way to check the individual power consumption of the CPUs only the power consumption of the entire machine. Which is not that useful anyway because the amount of power being consumed by a computer varies depending on the hardware.

**Comment by JakeRoggenbuck** [comment]
That's alright, I wasn't sure if there was a way to find it. I'll probably close the issue, or put a pin in it or something, if we happen to stumble upon a way to calculate that info in the future.

**Comment by Shuzhengz** [comment]
I think u can check the current TDP on Intel CPUs

**Comment by JakeRoggenbuck** [comment]
Oh! That would be cool!

**Comment by Shuzhengz** [comment]
Try `cat /sys/class/power_supply/BAT0/power_now ` and see if it works on your system

It should return the current CPU TDP in µW

**Comment by JakeRoggenbuck** [comment]
It has these two 
![image](https://user-images.githubusercontent.com/35516367/133004552-9f44ba0c-a944-4521-9497-982dc4e4d0f5.png)

**Comment by Shuzhengz** [comment]
yeaaa the voltage is the voltage, u can calculate the TDP if you have current, but without that its useless because Intel has disabled voltage modification to patch a hardware venerability
The charge is just how much juice you've got left
Try power_now on your desktop

**Comment by JakeRoggenbuck** [comment]
It doesn't look like it's there

**Comment by Shuzhengz** [comment]
sad
hmm maybe there's some other ways to find it

**Comment by Shuzhengz** [comment]
lol i spent like 2 hrs digging, only found this
![image](https://user-images.githubusercontent.com/43869232/149646390-c80e49ab-acaa-4f4e-9080-31ae3970f9da.png)

**Comment by Camerooooon** [comment]
I figured it out we can use ohms law by taking .../BAT0/voltage_now and multiplying that by .../BAT0/current_now to find the wattage

**Comment by Camerooooon** [comment]
See https://github.com/polybar/polybar/blob/master/src/modules/battery.cpp

**Comment by Shuzhengz** [comment]
> I figured it out we can use ohms law by taking .../BAT0/voltage_now and multiplying that by .../BAT0/current_now to find the wattage
> 
> 

Yea but that's what I was saying before, we only have the voltage and the charge, current is missing

**Comment by Camerooooon** [comment]
> > I figured it out we can use ohms law by taking .../BAT0/voltage_now and multiplying that by .../BAT0/current_now to find the wattage
> > 
> > 
> 
> Yea but that's what I was saying before, we only have the voltage and the charge, current is missing

current_now is the current

**Comment by Shuzhengz** [comment]
> > > I figured it out we can use ohms law by taking .../BAT0/voltage_now and multiplying that by .../BAT0/current_now to find the wattage
> > > 
> > > 
> > 
> > Yea but that's what I was saying before, we only have the voltage and the charge, current is missing
> 
> current_now is the current

We don't have current_now on our laptops, only voltage_now and charge_now

**Comment by Camerooooon** [comment]
> > > > I figured it out we can use ohms law by taking .../BAT0/voltage_now and multiplying that by .../BAT0/current_now to find the wattage
> > > > 
> > > > 
> > > 
> > > Yea but that's what I was saying before, we only have the voltage and the charge, current is missing
> > 
> > current_now is the current
> 
> We don't have current_now on our laptops, only voltage_now and charge_now

I have it on my computer strange...

**Comment by Shuzhengz** [comment]
Yeah, might be something with the newer Intel systems' BIOS (there was a vulnerability regarding to IccMax and VCCIN a few months ago on intel 8th - 10th gen processors, maybe the manufactures released BIOS that locked the current down to patch it)

**Comment by JakeRoggenbuck** [comment]
Ahh, hm, we can check if it exists and if it does, calculate it, if not, then don't but I'm not sure

**Comment by Shuzhengz** [comment]
> It has these two
> ![image](https://user-images.githubusercontent.com/35516367/133004552-9f44ba0c-a944-4521-9497-982dc4e4d0f5.png)

em I thought u already checked it lol

**Comment by Shuzhengz** [comment]
It doesn’t exist on my new laptop (X1 Extreme running i7-8850H), but it does for my very old laptop (T60 running Core 2 Duo)

**Comment by JakeRoggenbuck** [comment]
Yee, we could have something in code saying
```c++
if voltage_now.exists() && charge_now.exists() {
    then calculate 
} else {
    don't
}
```

**Comment by Shuzhengz** [comment]
Sure, but we probably can't use TDP to do much useful things (like auto limit clockspeed based on TDP) because it seems like a lot of computers don't have it)

**Comment by JakeRoggenbuck** [comment]
yea, we can put it on the back burner and not worry about it unless someone wants to work on it

**Comment by Shuzhengz** [comment]
yee
